src/frontends/pytorch/src/transforms/prim_list_construct_pad.cpp

```diff
@@ -0,0 +1,118 @@
+#include "openvino/op/pad.hpp"
+#include "openvino/op/range.hpp"
+#include "openvino/op/shape_of.hpp"
+#include "openvino/op/subtract.hpp"
+#include "openvino/op/util/framework_node.hpp"
+#include "openvino/pass/pattern/matcher.hpp"
+#include "openvino/pass/pattern/op/wrap_type.hpp"
+#include "pt_framework_node.hpp"
+#include "utils.hpp"
+
+namespace ov {
+namespace frontend {
+namespace pytorch {
+namespace pass {
+
+PrimListConstructPadReplacer::PrimListConstructPadReplacer() {
+    auto pad_op = ov::pass::pattern::wrap_type<ov::op::util::FrameworkNode>();
+    ov::matcher_pass_callback callback = [](ov::pass::pattern::Matcher& m) {
+        auto pad_op = cast_fw_node(m.get_match_root(), "aten::pad");
+        if (!pad_op) {
+            return false;
+        }
+        auto pt_node = std::dynamic_pointer_cast<PtFrameworkNode>(pad_op);
+        auto input_node = pad_op->input_value(0).get_node_shared_ptr();
+        auto padding = pad_op->input_value(1).get_node_shared_ptr();
+        auto pad_values = concat_list_construct(padding);
+        std::string mode = "constant";
+        auto minus_one = ov::op::v0::Constant::create(element::i64, Shape{}, {-1});
+        auto zero = ov::op::v0::Constant::create(element::i64, Shape{}, {0});
+        auto zero_f = ov::op::v0::Constant::create(element::f32, Shape{}, {0});
+        auto minus_two = ov::op::v0::Constant::create(element::i64, Shape{}, {-2});
+        auto input_shape = std::make_shared<ov::op::v3::ShapeOf>(input_node);
+        auto input_rank = std::make_shared<ov::op::v3::ShapeOf>(input_shape);
+        auto pad_size_1d = std::make_shared<ov::op::v3::ShapeOf>(pad_values);
+        auto pad_size = std::make_shared<ov::op::v0::Squeeze>(pad_size_1d, zero);
+        auto start_pad_begins = std::make_shared<ov::op::v1::Add>(pad_size, minus_two);
+        auto start_pad_ends = std::make_shared<ov::op::v1::Add>(pad_size, minus_one);
+        auto pad_begins_range =
+            std::make_shared<ov::op::v4::Range>(start_pad_begins, minus_one, minus_two, element::i64);
+        auto pad_ends_range = std::make_shared<ov::op::v4::Range>(start_pad_ends, zero, minus_two, element::i64);
+        auto pad_begins = std::make_shared<ov::op::v8::Gather>(pad_values, pad_begins_range, zero);
+        auto pads_short_len = std::make_shared<ov::op::v3::ShapeOf>(pad_begins);
+        auto pad_ends = std::make_shared<ov::op::v8::Gather>(pad_values, pad_ends_range, zero);
+        auto pads_diff = std::make_shared<ov::op::v1::Subtract>(input_rank, pads_short_len);
+        auto pads_remaining = std::make_shared<ov::op::v3::Broadcast>(zero, pads_diff);
+        auto pads_remaining_c = std::make_shared<ov::op::v1::ConvertLike>(pads_remaining, pad_begins);
+        auto pad_begins_full = std::make_shared<ov::op::v0::Concat>(OutputVector{pads_remaining_c, pad_begins}, 0);
+        auto pad_ends_full = std::make_shared<ov::op::v0::Concat>(OutputVector{pads_remaining_c, pad_ends}, 0);
```

Comment:
I think we can create the common helper to compute `pad_begins_full` and `pad_ends_full`. This way we can reduce some amount of code. Please consider.